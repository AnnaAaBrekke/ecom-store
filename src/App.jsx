import "./App.css";
import { Routes, Route } from "react-router-dom";
import Layout from "./components/Layout";
import Checkout from "./pages/checkout";
import Contact from "./pages/contact";
import Home from "./pages/home";
import ProductPage from "./pages/product";

function App() {
  return (
    <Layout>
      <Routes>
        <Route path="/" element={<Home />} /> {/* Home Page */}
        <Route path="/contact" element={<Contact />} /> {/* Contact Page */}
        <Route path="/checkout" element={<Checkout />} /> {/* Checkout Page */}
        <Route path="product/:id" element={<ProductPage />} />
        <Route path="*" element={<h1>404 - Page Not Found</h1>} />{" "}
        {/* 404 Page */}
      </Routes>
    </Layout>
  );
}

export default App;
