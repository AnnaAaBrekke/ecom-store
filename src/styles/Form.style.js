import styled from "styled-components";

export const StyledForm = styled.div`
  display: flex;
  flex-direction: column;
  justify-items: center;
  padding: 2.5rem;
  width: 340px;
  background: ${({ theme }) => theme.colors.accent};
  border-radius: ${({ theme }) => theme.borderRadius};
`;

export const FormGroup = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Label = styled.label`
  font-size: 1rem;
  font-weight: bold;
  margin-bottom: 0.3rem;
`;

export const Input = styled.input`
  padding: 0.75rem;
  border: 2px solid ${({ theme }) => theme.colors.accent};
  border-radius: ${({ theme }) => theme.borderRadius};
  background: ${({ theme }) => theme.colors.backgroundLight};
  transition: all 0.3s ease;
  font-size: 1rem;

  &:focus {
    border-color: ${({ theme }) => theme.colors.primary};
    outline: none;
    box-shadow: 0 0 8px ${({ theme }) => theme.colors.primaryLight};
  }
`;

export const TextArea = styled.textarea`
  padding: 0.75rem;
  border: 2px solid ${({ theme }) => theme.colors.accent};
  border-radius: ${({ theme }) => theme.borderRadius};
  background: ${({ theme }) => theme.colors.backgroundLight};
  transition: all 0.3s ease;
  font-size: 1rem;
  height: 120px;
  resize: vertical;
  line-height: 1.4;

  &:focus {
    border-color: ${({ theme }) => theme.colors.primary};
    outline: none;
    box-shadow: 0 0 8px ${({ theme }) => theme.colors.primaryLight};
  }
`;
