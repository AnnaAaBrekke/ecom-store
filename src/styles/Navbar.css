.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 12px 24px;
  background-color: #4a90e2; /* Soft blue */
  color: #fff;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1); /* Light shadow */
  border-radius: 8px;
}

.nav-logo {
  font-size: 24px;
  font-weight: bold;
}

.logo-link {
  color: #fff;
  text-decoration: none;
  transition: color 0.2s ease-in-out;
}

.logo-link:hover {
  color: #ffd700; /* Gold */
}

.nav-links {
  list-style: none;
  display: flex;
  gap: 20px;
  margin: 0;
  padding: 0;
}

.nav-item {
  font-size: 16px;
}

.nav-link {
  color: #fff;
  text-decoration: none;
  font-weight: 500;
  padding: 8px 12px;
  border-radius: 4px;
  transition: background-color 0.2s ease-in-out, color 0.2s ease-in-out;
}

.nav-link:hover {
  background-color: #ffd700;
  color: #4a4a4a;
}
