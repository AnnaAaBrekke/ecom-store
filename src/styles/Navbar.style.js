import styled from "styled-components";
import { Link } from "react-router-dom";

export const NavbarContainer = styled.nav`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 12px 24px;
  background-color: ${({ theme }) => theme.colors.layout};
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1); /* Light shadow */
  border-radius: ${({ theme }) => theme.borderRadiusSecondary};
  font-family: ${({ theme }) => theme.typography.fontFamily};
`;

export const LogoLink = styled(Link)`
  text-decoration: none;
  transition: color 0.2s ease-in-out;

  &:hover {
    color: #ffd700;
  }
`;

export const LogoImage = styled.img`
  width: 80px; /* Adjust size */
  height: auto;
  transition: opacity 0.3s ease-in-out;

  &:hover {
    opacity: 0.8; /* Subtle hover effect */
  }
`;

export const NavLinks = styled.ul`
  list-style: none;
  display: flex;
  gap: 20px;
  margin: 0;
  padding: 0;
`;

export const NavItem = styled.li`
  font-size: 16px;
  display: flex;
  align-items: center;
`;

export const NavLinkStyled = styled(Link)`
  color: ${({ theme }) => theme.colors.text};
  text-decoration: none;
  font-weight: 600;
  font-size: 20px;
  font-family: ${({ theme }) => theme.typography.fontFamily};
  padding: 8px 12px;
  border-radius: 4px;
  transition: background-color 0.2s ease-in-out, color 0.2s ease-in-out;

  &:hover {
    background-color: #ffd700;
    color: #4a4a4a;
  }
`;
