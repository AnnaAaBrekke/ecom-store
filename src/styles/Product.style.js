import styled from "styled-components";
import { Link } from "react-router-dom";

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  justify-content: center;
  align-items: center;
  gap: 2rem;
  margin: 1rem;
  box-sizing: border-box;
  overflow: hidden;
`;

export const Message = styled.div`
  text-align: center;
  font-size: 1.2rem;
  color: #555;
  margin-top: 20px;
`;

export const Card = styled.div`
  border: 1px solid ${({ theme }) => theme.colors.accent};
  padding: 12px;
  border-radius: ${({ theme }) => theme.borderRadius};
  background: ${({ theme }) => theme.colors.background};
  transition: transform 0.3s ease;
  height: auto;
  width: 90%;
  box-shadow: ${({ theme }) => theme.shadows.soft};
  overflow: hidden;
  margin: 0 auto;

  &:hover {
    transform: translateY(-5px); /* Subtle floating effect */
    box-shadow: ${({ theme }) => theme.shadows.strong};
  }
`;

export const ProductImage = styled.img`
  width: 100%;
  height: 240px;
  object-fit: cover;
  border-radius: ${({ theme }) => theme.borderRadius};
`;

export const CardBody = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  flex-grow: 1;
  width: 100%;
  padding: 10px;
  text-align: left;
`;

export const Title = styled.h2`
  font-size: 1.1rem;
  color: ${({ theme }) => theme.colors.text};
  font-family: ${({ theme }) => theme.typography.headingFont};
  margin-bottom: 8px;
`;

export const Description = styled.p`
  font-size: 0.9rem;
  color: ${({ theme }) => theme.colors.text};
  flex-grow: 1; /* Prevents overflow */
  overflow: hidden;
  text-overflow: ellipsis;
  max-height: 50px;
  white-space: nowrap;
`;

export const PriceContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 8px;
  margin-top: 6px;
  flex-wrap: wrap;
  justify-content: left;
`;

export const Price = styled.span`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 18px;
  font-weight: bold;
`;

export const OriginalPrice = styled.span`
  text-decoration: line-through;
  color: ${({ theme }) => theme.colors.error};
  font-size: 14px;
  opacity: 0.7;
`;

export const DiscountTag = styled.span`
  background-color: ${({ theme }) => theme.colors.success};
  color: white;
  font-size: 0.9rem;
  font-weight: bold;
  padding: 3px 8px;
  border-radius: 5px;
  display: inline-block;
  text-transform: uppercase;
`;

export const Rating = styled.div`
  font-size: 12px;
  color: ${({ theme }) => theme.colors.accent};
`;

export const Tags = styled.div`
  display: flex;
  justify-content: left;
  gap: 6px;
  flex-wrap: wrap;
  margin-top: 8px;
`;

export const Tag = styled.span`
  background-color: ${({ theme }) => theme.colors.accent};
  color: ${({ theme }) => theme.colors.secondary};
  padding: 4px 8px;
  border-radius: 4px;
  font-size: 0.8rem;
  margin-bottom: 16px;
`;

export const ViewButton = styled(Link)`
  width: 90%;
  padding: 10px;
  background: ${({ theme }) => theme.gradients.primaryGradient};
  color: ${({ theme }) => theme.colors.white};
  text-align: center;
  border-radius: ${({ theme }) => theme.borderRadius};
  text-decoration: none;
  font-weight: bold;
  transition: transform 0.2s ease, box-shadow 0.2s ease;
  box-shadow: ${({ theme }) => theme.shadows.soft};

  &:hover {
    transform: scale(1.05);
    box-shadow: ${({ theme }) => theme.shadows.strong};
  }
`;

// Product Single Page

export const AddToCartButton = styled.button`
  display: flex;
  align-items: center;
  gap: 10px;
  background: ${({ theme }) => theme.colors.primary};
  color: white;
  font-size: 1rem;
  font-weight: bold;
  padding: 12px 16px;
  border: none;
  border-radius: ${({ theme }) => theme.borderRadius};
  cursor: pointer;
  transition: 0.3s;

  &:hover {
    background: ${({ theme }) => theme.colors.secondary};
  }
`;
