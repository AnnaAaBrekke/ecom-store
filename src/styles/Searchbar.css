.search-suggestion-line {
  transition: background-color 0.2s ease;
}

.search-suggestion-line.active {
  background-color: yellow; /* Highlight color for active suggestion */
}

.search-suggestion-line:hover {
  background-color: #e8e8e8;
}
