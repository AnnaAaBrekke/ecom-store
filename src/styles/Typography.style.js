import styled from "styled-components";

export const Heading = styled.h1`
  font-size: 2.5rem;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.primary};
  text-align: center;
  margin-bottom: 15px;
  text-shadow: 3px 3px 10px rgba(0, 0, 0, 0.8);
`;

export const SubHeader = styled.h2`
  font-size: 1.5rem;
  font-weight: 600;
  color: ${({ theme }) => theme.colors.secondary};
  margin-bottom: 20px;
  text-align: center;
`;

export const Paragraph = styled.p`
  font-size: 1rem;
  font-weight: 400;
  color: ${({ theme }) => theme.colors.secondaryText || theme.colors.text};
  line-height: 1.6;
  margin-bottom: 15px;
  text-align: center;
  max-width: 800px;
`;
