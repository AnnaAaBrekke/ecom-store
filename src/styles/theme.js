export const theme = {
  colors: {
    primary: "#1B3A4B", // Deep Midnight for a Bold and Strong Feel
    secondary: "#455A64", // Fjord Blue for a Cool and Modern Accent
    layout: "#BCCACC",
    accent: "#BCCACC", // Cool Slate for Subtle Highlights
    background: "#ECEFF1", // Polar Gray for a Clean and Soft Look
    text: "#0D1B2A", // Nordic Black for High Contrast Readability
    success: "#4caf50", // Green for Success Messages
    error: "#ff3d3d", // Bright Red for Errors
    white: "#ffffff", // Classic White for Contrast
    black: "#000000", // Deep Black for Depth
  },

  gradients: {
    primaryGradient: "linear-gradient(135deg, #1B3A4B 0%, #455A64 100%)", // Smooth Midnight to Fjord Blue Gradient
  },

  typography: {
    fontFamily: "'Playfair Display', serif", // Elegant & Stylish Font
    fontSize: "16px",
    heading1: "32px",
  },

  borderRadius: "12px",

  shadows: {
    soft: "0px 4px 10px rgba(0, 0, 0, 0.1)",
    strong: "0px 6px 20px rgba(0, 0, 0, 0.2)",
  },
};
