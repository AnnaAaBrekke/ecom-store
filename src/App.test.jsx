// createRoot(document.getElementById("root")).render(
//   <StrictMode>
//     <ThemeProvider theme={theme}>
//       <BrowserRouter>
//         <App />
//       </BrowserRouter>
//     </ThemeProvider>
//   </StrictMode>
// );

import React from "react";
import "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import App from "./App";
import { theme } from "./styles/theme";
import { expect, it } from "vitest";

it("App component should match snapshot", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
