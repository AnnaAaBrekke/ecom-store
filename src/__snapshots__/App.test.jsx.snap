// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component should match snapshot 1`] = `null`;
